service-workers/service-worker/sandboxed-iframe-fetch-event.https.html

```diff
@@ -61,475 +61,355 @@
 // This should be controlled by a service worker.
 let sandboxed_same_origin_frame_by_header;
 
-promise_test(t => {
-  return service_worker_unregister_and_register(t, SCRIPT, SCOPE)
-    .then(function(registration) {
-      add_completion_callback(() => registration.unregister());
-      worker = registration.installing;
-      return wait_for_state(t, registration.installing, 'activated');
-    });
+promise_test(async t => {
+  add_completion_callback(async () => {
+    if (registration) await registration.unregister();
+  });
+
+  const registration = await service_worker_unregister_and_register(
+    t,
+    SCRIPT,
+    SCOPE
+  );
+
+  worker = registration.installing;
```

Comment:
bit confused here... |worker| looks unused? Should there be a global `let worker` be declared outside these promise_tests? If so, I'd prefer getResultsFromWorker take a different param like |targetWorker|.